### DISCUSSION TITLE
20230711 회의

### DISCUSSION THREAD

**Discussion by KKangTaee** [KKangTaee/HifeSurvival#36] [General]
## 완료 된 내용
* IncreaseStat 처리. 재화(골드)에 따른 스텟 업그레이드 시스템
* 캐릭터 선택창에서 모두 레디를 안해도 바로 인게임이 진입되는 이슈
* 아이템 장착 시 공격력이 비 이상적으로 증가하는 이슈 수정

## 확인 및 수정 할 내용
* 특정 몬스터 사망 시 사망 연출 안나오는 이슈 (클라)
* 타 플레이어 이동이 정지 했음에도 Idle 모션이 간혈적으로 나오는 이슈 (클라)
* 캐릭터 선택창에서 프로필 이미지 안나오는 이슈 (클라)

## 해야할 일
* 단계에 따른 순차적 리스폰 처리 (Phase Spawn)
* 게임시작, 종료처리 (클라, 서버)
* 파티클 매니저 구현 (클라)
* 아이템 성장 시스템 구현 (클라)
* 몬스터 AI 이동 (클라, 서버)
* 어드레서블 구현 (클라)
* 몬스터 제자리로 돌아갈 때. 체력회복
* 플로우 레이어 나누기 (서버)

**Comment by LeeHyungJoo** [comment]
몬스터 ID 0 ~ 9999
캐릭터 ID 10000 부터

**Comment by LeeHyungJoo** [comment]
#로비 씬

## 이전
C_JoinToGame
S_JoinToGame
CS_SelectHero
CS_ReadyToGame
S_LeaveToGame

## 리뉴얼
JoinToGameRequest 
동일
JoinToGameResponse 
동일

SelectHeroRequest
동일
SelectHeroResponse
동일 + 응답 결과 값.

ReadyToGameRequest
동일
ReadyToGameResponse 
동일 + 응답 결과 값. 


LobbyUpdate
타겟 id 


Add Info
*	Id
*	userName
*	heroKey


Delete Id
*	Id

Update Info
*	heroKey
*	ready

**Comment by LeeHyungJoo** [comment]
## 기본 통신 (Req, Res) 
세션ID 만 필요. 
패킷에서 Request 하는 부분 ID 다 삭제해야함. 
필요 없는 정보, UserId 로 획득한 ID 정보를 들고 
이를 세션과 키 / 밸류로 묶어서 사용. 
그래야, 약 네트워크 상황에서, 세션이 바뀌어도, 해당 밸류를 찾아서 연결 할 수 있음. 

## 게임 씬

### 이전
CS_Attack

### 리뉴얼
AttackRequest
DealtBroadcast (가칭, 전투 프레임 하나에 어떤 결과가 났는지 관리하는 패킷)

채력 갱신. (유닛 업데이트 패킷)
ID
증가/감소된 값. 
체력 결과 값. 
(이펙트 타입)
